src/alarms/notifications/App.js

```diff
@@ -59,50 +80,165 @@ class App extends Component {
       .then(data => {
         this.setState({
           isFetching: false,
-          total: data.count,
           alarms: data.results,
-          page: page
+          total: data.count
         });
+      })
+      .catch(error => {
+        console.error("Error:", error);
+      });
+  }
+
+  urlFromAlarm(alarm) {
+    if (alarm.url.includes("timeseriesalarms"))
+      return `/api/v3/timeseriesalarms/${alarm.uuid}/`; //(alarm.type==='RasterAlarm')
+    else return `/api/v3/rasteralarms/${alarm.uuid}/`;
+  }
+
+  activateAlarm(alarm) {
+    const { addNotification } = this.props;
+
+    fetch(this.urlFromAlarm(alarm), {
+      credentials: "same-origin",
+      method: "PATCH",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
```

Comment:
De fetch met die credentials, headers, aanroepen van JSON.stringify, vertalen van de response naar json... komt heel vaak weer terug, handiger als het een functie is.

Reply:
Okey, welke items zal ik allemaal parametriseren in deze functie?
De meest simpele approach zal zijn om het een asynchrone functie te maken en alleen de volgende te parametriseren:
- method
- body
Wat denk jij?

Reply:
Zie commit hieronder.
Is dit wat je in gedachten had